### PR TITLE
fix: Make auto-update work again, the lazy way

### PR DESCRIPTION
**Pull request by KazWolfe**
This pull request aims to at least get auto update working again.

Auto updates were recently broken per some PM rewrites. This (alongside a case where a certain custom-repo plugin was triggering a chat message on the login screen) caused Dalamud to attempt to auto update before the update list was properly calculated, thereby not doing anything but marking the auto-update phase as complete. 

While it is possible to (and we probably *should*) fix the underlying problem with PluginManager such that an auto update triggers after all repos are loaded, this is somewhat more work and probably won't be clean. Per goat, this may be a good opportunity to also handle some cases such as informing plugins that there may be an update present.

### CODE REVIEW

File: Dalamud/Game/ChatHandlers.cs [hunk 1]
```diff
@@ -6,6 +6,7 @@
 
 using CheapLoc;
 using Dalamud.Configuration.Internal;
+using Dalamud.Game.ClientState.JobGauge.Enums;
```
Comment (goaaats) [inline Dalamud/Game/ChatHandlers.cs:9]: What's this for?

File: Dalamud/Game/ChatHandlers.cs [hunk 2]
```diff
@@ -165,16 +171,25 @@ private void OnChatMessage(XivChatType type, uint senderId, ref SeString sender,
         if (clientState == null)
             return;
 
+        if (type == XivChatType.Notice)
+        {
+            if (!this.hasSeenLoadingMsg)
+                this.PrintWelcomeMessage();
+            
+            // FIXME (kazwolfe): This will only run on the first Notice received, meaning auto-updates won't happen
```
Comment (goaaats) [inline Dalamud/Game/ChatHandlers.cs:179]: I actually don't think we should change this. A lot of plugins are expecting to be updated when the character is logged in, and will show stuff like changelogs via ImGui or in chat.

File: Dalamud/Game/ChatHandlers.cs [hunk 3]
```diff
@@ -271,17 +286,22 @@ private void AutoUpdatePlugins()
         var notifications = Service<NotificationManager>.GetNullable();
 
         if (chatGui == null || pluginManager == null || notifications == null)
+        {
+            Log.Warning("Aborting auto-update because a required service was not loaded.");
             return;
+        }
 
         if (!pluginManager.ReposReady || !pluginManager.InstalledPlugins.Any() || !pluginManager.AvailablePlugins.Any())
```
Comment (goaaats) [inline Dalamud/Game/ChatHandlers.cs:294]: We should retry immediately when this happens, once we saw the first notice.

**Review by goaaats — CHANGES_REQUESTED** [review]

File: Dalamud/Game/ChatHandlers.cs (hunk 1 above)
Comment (KazWolfe) [reply Dalamud/Game/ChatHandlers.cs]: derp.

File: Dalamud/Game/ChatHandlers.cs (hunk 2 above)
Comment (KazWolfe) [reply Dalamud/Game/ChatHandlers.cs]: Hrm. Seems strange to defer updates like this but in the interest of compat I'll drop this for now.

Plugins imo shouldn't expect to update anywhere in particular, and I'd even argue that if we're auto-updating plugins, we shouldn't even load the old version. I'm sure this is one of those "sounds easy on paper but painful in practice" things, so it's not particularly worth digging for now.

If we *do* want auto-updates to fire after character login, we should still move this out of ChatHandlers - nothing is stopping a plugin from sending a `Notice` on the login screen (which is how we got into this mess in the first place).

File: Dalamud/Game/ChatHandlers.cs (hunk 3 above)
Comment (KazWolfe) [reply Dalamud/Game/ChatHandlers.cs:284]: How would you like me to implement this?

Any future `Notice` will already attempt to re-trigger an auto update (as the `startedAutoUpdaingPlugins` bool hasn't been tripped yet). If you want the first Notice to still perform a retrigger I can make this method return a bool and retry from the caller, but that feels like we're just racing code at that point.

**Review by goaaats — APPROVED** [review]

**Review by goaaats — APPROVED** [review]

**Pull request merged**